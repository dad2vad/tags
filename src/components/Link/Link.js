import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class Link extends PureComponent {
  render() {
    return (
      <a
        href={this.props.url}
        data-tooltip={this.props.tooltip}
        target="_blank"
        rel="noopener noreferrer"
        className={classnames({ 'is-tooltip': this.props.tooltip }, this.props.className)}
      >
        {this.props.text}
        {this.props.children}
      </a>
    );
  }
}

Link.propTypes = {
  url: PropTypes.string.isRequired,
  text: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node,
  tooltip: PropTypes.string,
};

export default Link;
