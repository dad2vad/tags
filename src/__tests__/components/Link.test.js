import React from 'react';
import { shallow } from 'enzyme';
import Link from '../../components/Link';

describe('Link', () => {
  it('renders correctly', () => {
    const component = shallow(
      <Link url="https://frameworks.now.sh/" text="Comparison of Front-End Frameworks" />
    );
    expect(component).toMatchSnapshot();
  });
});
