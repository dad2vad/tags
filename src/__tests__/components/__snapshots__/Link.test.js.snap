// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className=""
  href="https://frameworks.now.sh/"
  rel="noopener noreferrer"
  target="_blank"
>
  Comparison of Front-End Frameworks
</a>
`;
